File: Leetcode_solutions/Maximum Depth of Binary Tree.js
var maxDepth = function(root) {
  if (!root) return 0;
  let max = 0;
  function dfs(node, level) {
    if (node.left) {
      dfs(node.left, level + 1);
    }
    if (node.right) {
      dfs(node.right, level + 1);
    } else {
      max = Math.max(max, level);
    }
  }
  dfs(root, 1);
  return max;
};

File: Leetcode_solutions/Maximum_Depth_of_Binary_Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    def maxDepth(self, root):
        if root == None:
            return 0
        else:
            right = self.maxDepth(root.right)
            left = self.maxDepth(root.left)
            return max(left, right) + 1


s = Solution()
s.maxDepth([3, 9, 20, None, None, 15, 7])
# print()
